### PR TITLE
Animation research 

### PR DESCRIPTION
**Pull request by Berthalamew**
- defined the animation graph definition as well as the member classes of it and some of the member functions
- update definitions for c_animation_channel and c_interpolator_control
- fixes compilation issues with constant tagblocks and string ids

- this is beneficial since it allows us to read and access animation tags incase someone wants to use them later on in project

### CODE REVIEW

File: xlive/Blam/Common/Common.h [hunk 1]
```diff
@@ -42,7 +42,7 @@ static_assert (sizeof(STRUCT) == _SIZE, "Invalid size for struct (" #STRUCT")")
 static_assert (offsetof(STRUCT, FIELD) == OFFSET, #STRUCT " layout is invalid")
 
 #define TAG_BLOCK_SIZE_ASSERT(tagblock,size)\
-static_assert (sizeof(tagblock) == (size),"Invalid Size for TagBlock <" #tagblock ">")
+static_assert (sizeof(tagblock) == size,"Invalid Size for TagBlock <" #tagblock ">")
```
Comment (nukeulater) [inline xlive/Blam/Common/Common.h]: why?

Comment (Berthalamew) [reply xlive/Blam/Common/Common.h]: Was looking at the assert while working on the definitions and the brackets seemed unnecessary.
I can add them back if you want.

File: xlive/Blam/Engine/animations/animation_definitions.cpp [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+}
+
+e_animation_type c_model_animation::get_animation_type() const
+{
+	return this->type;
+}
+
+float c_model_animation::get_authored_duration() const
+{
+	return this->frame_count * 0.0333333333f;
```
Comment (nukeulater) [inline xlive/Blam/Engine/animations/animation_definitions.cpp]: `frame_count / 30.0f` or `frame_count * (1.0f / 30.0f)` would be more preferable

File: xlive/Blam/Common/Common.h (hunk 1 above)
Comment (nukeulater) [reply xlive/Blam/Common/Common.h]: macro pitfalls: https://gcc.gnu.org/onlinedocs/gcc-3.0.1/cpp_3.html#SEC26

File: xlive/Blam/Engine/animations/animation_definitions.cpp (hunk 2 above)
Comment (Berthalamew) [reply xlive/Blam/Engine/animations/animation_definitions.cpp]: changed to this->frame_count * (1.0f / 30.0f)

File: xlive/Blam/Common/Common.h (hunk 1 above)
Comment (Berthalamew) [reply xlive/Blam/Common/Common.h]: reverted back to before

**Pull request merged**